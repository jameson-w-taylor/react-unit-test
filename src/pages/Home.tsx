import { IonButton, IonContent, IonHeader, IonPage, IonTitle, IonToolbar, useIonModal } from '@ionic/react';
import MyModal, { MyModalProps } from './MyModal';
import './Home.css';

const Home: React.FC = () => {
  const [presentModal, dismissModal] = useIonModal(MyModal, {
    onDismiss: () => handleOnModalDismiss(),
  } as MyModalProps);

  const handleOnModalDismiss = () => {
    dismissModal();
  }

  return (
    <IonPage>
      <IonHeader>
        <IonToolbar>
          <IonTitle>Blank</IonTitle>
        </IonToolbar>
      </IonHeader>
      <IonContent fullscreen>
        <IonHeader collapse="condense">
          <IonToolbar>
            <IonTitle size="large">Blank</IonTitle>
          </IonToolbar>
        </IonHeader>
        <IonButton onClick={() => presentModal()} data-testid="openModalButton">
          Open Modal
        </IonButton>
      </IonContent>
    </IonPage>
  );
};

export default Home;
