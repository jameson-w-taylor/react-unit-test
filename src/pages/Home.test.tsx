import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import { ionFireEvent as fireEvent } from '@ionic/react-test-utils';
import Home from './Home';

describe('<Home />', () => {
  it('renders consistently', async () => {
    const { baseElement } = render(<Home />);
    /**
     * Current snapshot contains no CSS classes normally added by Ionic at runtime
     * If you uncomment the 500ms delay below you'll see that the snapshot test will fail and output shows received DOM contains CSS classes
     */
    // await new Promise(resolve => {
    //   setTimeout(() => {
    //     resolve('wait');
    //   }, 500);
    // });
    expect(baseElement).toMatchSnapshot();
  });

  describe('open my modal', () => {
    it('displays my modal', async () => {
      const { getByText, getByTestId } = render(<Home />);
      const button = getByTestId(/openModalButton/) as HTMLIonButtonElement;
      fireEvent.click(button);
      try {
        // Will retry finding IonModal title for 1000ms before failing
        // https://testing-library.com/docs/dom-testing-library/api-async#waitfor
        await waitFor(() => expect(getByText('My Modal')).toBeDefined());
      } catch (e) {
        // Will end up here if IonModal title is not found, notice modal content is missing in DOM output
        // https://testing-library.com/docs/queries/about#screendebug
        screen.debug();
      }
    });
  });
});
