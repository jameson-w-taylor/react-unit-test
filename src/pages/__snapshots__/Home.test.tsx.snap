// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders consistently 1`] = `
<body>
  <div>
    <div
      class="ion-page"
    >
      <ion-header>
        <!---->
        <ion-toolbar>
          <ion-title>
            Blank
          </ion-title>
        </ion-toolbar>
      </ion-header>
      <ion-content
        fullscreen="true"
      >
        <ion-header
          collapse="condense"
        >
          <!---->
          <ion-toolbar>
            <ion-title
              size="large"
            >
              Blank
            </ion-title>
          </ion-toolbar>
        </ion-header>
        <ion-button
          data-testid="openModalButton"
        >
          Open Modal
        </ion-button>
      </ion-content>
    </div>
  </div>
</body>
`;
